crates/core/tedge/src/system_services/managers/config.rs

```diff
@@ -0,0 +1,120 @@
+use crate::system_services::SystemConfigError::{ConfigFileNotFound, InvalidSyntax};
+use crate::system_services::*;
+use serde::Deserialize;
+use std::fs;
+use std::path::PathBuf;
+
+const SERVICE_CONFIG_FILE: &str = "system.toml";
+
+#[derive(Deserialize, Debug, PartialEq)]
+pub struct SystemConfig {
+    pub name: String,
+    pub is_available: Vec<String>,
+    pub restart: Vec<String>,
+    pub stop: Vec<String>,
+    pub enable: Vec<String>,
+    pub disable: Vec<String>,
+    pub is_active: Vec<String>,
+}
+
+impl Default for SystemConfig {
+    fn default() -> Self {
+        Self {
+            name: "systemd".to_string(),
+            is_available: vec!["/bin/systemctl".into(), "--version".into()],
+            restart: vec!["/bin/systemctl".into(), "restart".into(), "{}".into()],
+            stop: vec!["/bin/systemctl".into(), "stop".into(), "{}".into()],
+            enable: vec!["/bin/systemctl".into(), "enable".into(), "{}".into()],
+            disable: vec!["/bin/systemctl".into(), "disable".into(), "{}".into()],
+            is_active: vec!["/bin/systemctl".into(), "is-active".into(), "{}".into()],
+        }
+    }
+}
+
+impl SystemConfig {
+    pub fn new(config_root: PathBuf) -> Self {
+        match Self::try_new(config_root) {
+            Ok(config) => config,
+            Err(err) => {
+                eprintln!("{:?}", err);
+                Self::default()
```

Comment:
This needs to be improved:

* The default is to be used only when no config file is provided, not in case there is another kind of error.
* When the default is used, a warning must tell it.